src/foam/u2/wizard/BaseWizardlet.js

```diff
@@ -312,11 +312,19 @@ foam.CLASS({
       if ( this.data ) this.data = this.data.clone(this.__subSubContext__);
     },
     function createWizardletSectionsFromModel_() {
+      // This method fails if the wizardlet's data is undefined
+      if ( ! this.data ) {
+        if ( ! this.of ) return [];
+
+        this.warn('initializing wizardlet data to initialize sections');
+        this.data = this.of.create();
+      }
+
       // Internal method used by SECTIONS.factory
       var sections = this.AbstractSectionedDetailView.create({
         of: this.of,
       }, this).sections
-        .filter(section => this.defaultSections.includes(section.name))
+        .filter(section => this.defaultSections.includes('' + section.name))
```

Comment:
Is the '' + necessary? Isn't section already a String?

Reply:
`section.name` can be `undefined`

longer explanation, if desired:
> `this.defaultSections` is populated using this same method if it wasn't set in a journal, however because it's a `StringArray` every value gets forced into a string. For the section containing properties that aren't in a named section, `section.name` is `undefined` rather than `''` or `'undefined'`

Reply:
So you're expecting undefined to be converted into 'undefined'? Better to add a comment to the code so that someone doesn't remove the "'' +" sometime in the future.